### DISCUSSION TITLE
Middleware affecting getInitialProps after 12.2

### DISCUSSION THREAD

**Discussion by joaogarin** [vercel/next.js#38917] [Help]
hello,

I have an app that has a _app.js with a `getInitialProps`, but also some other pages that still have `getInitialProps` implemented. 

After updating to Next.js 12.2 we are noticing that the presence of the middleware file makes it so that these `getInitialProps` run both on client and server always (first the sever then the client), removing the middleware file restores the old behaviour where the `getInitialProps` runs only on the client when doing a client transition and on the server when doing a server reload 

this feels a bit at odds with the documentation on `getInitialProps` https://nextjs.org/docs/api-reference/data-fetching/get-initial-props  "For the initial page load, `getInitialProps` will run on the server only. `getInitialProps` will then run on the client when navigating to a different route via the next/link component or by using next/router" -> The page in question does not implement `getServersideProps` (but I do have a `getInitialProps` in `_app.js` as described above)

My question is : Is this intentional behaviour? that adding a middleware affects how the `getInitialProps` function behaves?  And was this meant to be introduced in 12.2? I understand that middleware was not stable before and so there are changes to it that were to be expected but since this behaviour is not documented I was wondering if this is indeed intentional or not.

Here are 2 videos of a bare bones next app demonstrating the issue : 

#### Without any middleware

Without any middleware the behaviour we can see is that `getInitialProps` works as documented, when doing a server load it runs on th server and when doing a client transition it runs on the client 

https://user-images.githubusercontent.com/2866604/180443142-d664c9eb-490a-4319-b185-02ffefff1ebe.mp4

#### With middleware

With middlware we can see that for the same app the behaviour changes, and it now runs always on server and client even when doing client transitions (where previously it would only run on the client)

https://user-images.githubusercontent.com/2866604/180443276-b6176885-0fab-43e5-95a0-35b8e413346e.mp4

Here is the repo reproducing the problem 

https://github.com/joaogarin/middleware-gip

Thank you!

**Comment by antoinelin** [comment]
Hello I'm having the same problem with `getStaticProps` can't succeed to get `pageProps` within `_app` when I navigate using the router. We started to talk about this issue on the community Discord. Sorry to bother you @timneutkens but is this something we can move as an issue? 🙏🏻
